core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataDeleteIndexService.java

```diff
@@ -93,13 +105,23 @@ public ClusterState execute(final ClusterState currentState) {
                 MetaData.Builder metaDataBuilder = MetaData.builder(meta);
                 ClusterBlocks.Builder clusterBlocksBuilder = ClusterBlocks.builder().blocks(currentState.blocks());
 
+                final IndexGraveyard.Builder graveyardBuilder = IndexGraveyard.builder(metaDataBuilder.indexGraveyard());
+                final int previousGraveyardSize = graveyardBuilder.tombstones().size();
                 for (final Index index : indices) {
                     String indexName = index.getName();
                     logger.debug("[{}] deleting index", index);
                     routingTableBuilder.remove(indexName);
                     clusterBlocksBuilder.removeIndexBlocks(indexName);
                     metaDataBuilder.remove(indexName);
+                    graveyardBuilder.addTombstone(index); // add the index tombstone to the cluster state
                 }
+                final int numPurged = graveyardBuilder.purge(SETTING_MAX_TOMBSTONES.get(settings)); // purge tombstone entries if needed
```

Comment:
So I'm wondering about this, if we know up front the number of tombstones that are added, do we need a max tombstone setting and `.purge` method at all? Couldn't this just pop the number of deleted indices' tombstones off of the front of the list?

The .purge makes it sound like the two are decoupled, but since they're not, would it be simpler to not deal with purging? (or, alternatively, make it decoupled)